File: Mythical Escape/Assets/My Stuff/Scripts/Door2.cs
using UnityEngine;
using System.Collections;

public class Door2 : MonoBehaviour
{
    private GameObject door;
    private SwitchPuzzle switchesPuzzle;

    void Awake()
    {
        door = GameObject.FindGameObjectWithTag("SF_Door2");
        switchesPuzzle = GameObject.FindObjectOfType<SwitchPuzzle>();
    }

    void OnTriggerEnter(Collider collider)
    {
        if (collider.CompareTag("Player") && switchesPuzzle.solved())
        {
            door.GetComponent<Animation>().Play("open");
        }
    }

    void OnTriggerExit(Collider collider)
    {
        if (collider.CompareTag("Player") && switchesPuzzle.solved())
        {
            door.GetComponent<Animation>().Play("close");
        }
    }
}

File: Mythical Escape/Assets/Door2.js
﻿#pragma strict

function OnTriggerEnter (obj : Collider) {
    var thedoor = gameObject.FindWithTag("SF_Door2");
    thedoor.GetComponent.<Animation>().Play("open");
}

function OnTriggerExit (obj : Collider) {
    var thedoor = gameObject.FindWithTag("SF_Door2");
    thedoor.GetComponent.<Animation>().Play("close");
}